core/src/test/java/apoc/spatial/GeocodeTest.java

```diff
@@ -34,8 +45,95 @@ public void testWrongUrlButViaOtherProvider() throws Exception {
         testGeocodeWithThrottling("osm", false, 
                 map("url", "https://api.opencagedata.com/geocode/v1/json?q=PLACE&key=KEY111"));
     }
+
+    @Test
+    public void testGeocodeWithBlockedAddressWithApocConf() {
+        final String geocodeBaseConfig = Geocode.PREFIX + "." + "opencage";
+        apocConfig().setProperty(geocodeBaseConfig + ".key", "myKey");
+        
+        Stream.of("https", "http", "ftp").forEach(protocol -> {
+            final String nonBlockedUrl = String.format(URL_FORMAT, protocol, NON_BLOCKED_ADDRESS);
+            final String nonBlockedReverseUrl = String.format(REVERSE_URL_FORMAT, protocol, NON_BLOCKED_ADDRESS);
+            
+            final String geocodeConfigUrl = geocodeBaseConfig + ".url";
+            final String geocodeConfigReverseUrl = geocodeBaseConfig + ".reverse.url";
+            
+            // check that if either url or reverse address are blocked 
+            // respectively the apoc.spatial.geocode and the apoc.spatial.reverseGeocode procedure fails
+            apocConfig().setProperty(geocodeConfigUrl, nonBlockedUrl);
+            apocConfig().setProperty(geocodeConfigReverseUrl, String.format(REVERSE_URL_FORMAT, protocol, BLOCKED_ADDRESS));
+
+            assertGeocodeFails(true, BLOCKED_ERROR);
+            
+            apocConfig().setProperty(geocodeConfigUrl, String.format(URL_FORMAT, protocol, BLOCKED_ADDRESS));
+            apocConfig().setProperty(geocodeConfigReverseUrl, nonBlockedReverseUrl);
+            
+            assertGeocodeFails(false, BLOCKED_ERROR);
+
+            // check that if neither url nor reverse url are blocked 
+            // the procedures continue the execution (in this case by throwing a `java.net` tException)
+            apocConfig().setProperty(geocodeConfigUrl, nonBlockedUrl);
+            apocConfig().setProperty(geocodeConfigReverseUrl, nonBlockedReverseUrl);
+            
+            assertGeocodeFails(false, JAVA_NET_EXCEPTION);
+            assertGeocodeFails(true, JAVA_NET_EXCEPTION);
+        });
+    }
```

Comment:
This test is taking about 1 minute to run. As if the test below. This seems a little bit excessive for these tests, do you think there is room to bring their execution down to ~few seconds?

Reply:
I speeded up the queries a lot by changing the `NON_BLOCKED_ADDRESS` value from `123.456.7.8` to `localhost`. This way you still get a java.net exception but faster (because it's not a socketTimeoutException but a Connection refused)